stl/inc/format

```diff
@@ -0,0 +1,2734 @@
+        case _Basic_format_arg_type::_None:
+            return basic_format_arg<_Context>{};
+        case _Basic_format_arg_type::_Int_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<int>(_Arg_storage)};
+        case _Basic_format_arg_type::_UInt_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<unsigned int>(_Arg_storage)};
+        case _Basic_format_arg_type::_Long_long_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<long long>(_Arg_storage)};
+        case _Basic_format_arg_type::_ULong_long_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<unsigned long long>(_Arg_storage)};
+        case _Basic_format_arg_type::_Bool_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<bool>(_Arg_storage)};
+        case _Basic_format_arg_type::_Char_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<_CharType>(_Arg_storage)};
+        case _Basic_format_arg_type::_Float_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<float>(_Arg_storage)};
+        case _Basic_format_arg_type::_Double_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<double>(_Arg_storage)};
+        case _Basic_format_arg_type::_Long_double_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<long double>(_Arg_storage)};
+        case _Basic_format_arg_type::_Pointer_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<const void*>(_Arg_storage)};
+        case _Basic_format_arg_type::_CString_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<const _CharType*>(_Arg_storage)};
+        case _Basic_format_arg_type::_String_type:
+            return basic_format_arg<_Context>{_Get_value_from_memory<basic_string_view<_CharType>>(_Arg_storage)};
+        case _Basic_format_arg_type::_Custom_type:
+            return basic_format_arg<_Context>{
+                _Get_value_from_memory<typename basic_format_arg<_Context>::handle>(_Arg_storage)};
+        default:
+            _STL_ASSERT(false, "Invalid basic_format_arg type");
+            return basic_format_arg<_Context>{};
+        }
+    }
+
+private:
+    size_t _Num_args              = 0;
+    const unsigned char* _Storage = nullptr;
+};
+
+// clang-format off
+template <class _Out, class _CharT>
+    requires output_iterator<_Out, const _CharT&>
+class basic_format_context {
+    // clang-format on
+private:
+    _Out _OutputIt;
+    basic_format_args<basic_format_context> _Args;
+    locale _Loc;
+
+public:
+    using iterator  = _Out;
+    using char_type = _CharT;
+
+    template <class _Ty>
+    using formatter_type = formatter<_Ty, _CharT>;
+
+    constexpr basic_format_context(
+        _Out _OutputIt_, basic_format_args<basic_format_context> _Ctx_args, const locale& _Loc_)
+        : _OutputIt(_STD move(_OutputIt_)), _Args(_Ctx_args), _Loc(_Loc_) {}

```

Comment:
Should we move the `_Ctx_args`?
```suggestion
        : _OutputIt(_STD move(_OutputIt_)), _Args(_STD move(_Ctx_args)), _Loc(_Loc_) {}
```

Reply:
It contains a `size_t` and non-owning `const unsigned char*` if I understand the code correctly, so it looks like it doesn't benefit from move semantics.

Reply:
I looked at other occurrences and none of them are moved. I think no change is necessary here.